torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2015,6 +2015,27 @@ def sample_inputs_logsumexp(self, device, dtype, requires_grad):
 
     return tuple(samples)
 
+def sample_inputs_like_fns(self, device, dtype, requires_grad, full_like=False):
+    inputs = (
+        ((), {}),
+        ((S, S), {}),
+        ((0, S, 0), {}),
+        ((S,), {'dtype': torch.double}),
+        ((S,), {'device': 'cpu'}),
+        ((S,), {'dtype': torch.double, 'device': 'cpu'}),
+    )
+    samples = []
+    for shape, kwargs in inputs:
+        t = make_tensor(shape, device, dtype,
+                        low=None, high=None,
+                        requires_grad=requires_grad)
+        if full_like:
+            samples.append(SampleInput(t, args=(123,), kwargs=kwargs))
+        else:
+            samples.append(SampleInput(t, kwargs=kwargs))
```

Comment:
I dislike the special casing here for a single op. I would prefer to have a separate sample inputs function for it instead:

```python
def sample_inputs_full_like(*args, fill_value=123, **kwargs):
    sample_inputs = []
    for sample_input in sample_inputs_like_fns(*args, **kwargs):
        sample_input.args = (fill_value,)
        sample_inputs.append(sample_input)
    return sample_inputs
```